ui/ducks/ens.js

```diff
@@ -0,0 +1,183 @@
+      } else {
+        log.error(reason);
+        state.error = ENS_UNKNOWN_ERROR;
+      }
+    },
+    processEnsResult: (state, action) => {
+      state.resolution = action.payload;
+      if (state.resolution === BURN_ADDRESS) {
+        state.error = ENS_NO_ADDRESS_FOR_NAME;
+      }
+      if (state.resolution === ZERO_X_ERROR_ADDRESS) {
+        state.error = ENS_REGISTRATION_ERROR;
+      }
+      if (
+        isValidDomainName(state.resolution) &&
+        isConfusing(state.resolution)
+      ) {
+        state.warning = CONFUSING_ENS_ERROR;
+      }
+    },
+    enableEnsLookup: (state, action) => {
+      state.stage = 'INITIALIZED';
+      state.error = null;
+      state.resolution = null;
+      state.warning = null;
+      state.network = action.payload;
+    },
+    disableEnsLookup: (state) => {
+      state.stage = 'NO_NETWORK_SUPPORT';
+      state.error = ENS_NOT_SUPPORTED_ON_NETWORK;
+      state.warning = null;
+      state.resolution = null;
+      state.network = null;
+    },
+    resetResolution: (state) => {
+      state.resolution = null;
+      state.warning = null;
+      state.error =
+        state.stage === 'NO_NETWORK_SUPPORT'
+          ? ENS_NOT_SUPPORTED_ON_NETWORK
+          : null;
+    },
+  },
+  extraReducers: (builder) => {
+    builder.addCase(CHAIN_CHANGED, (state, action) => {
+      if (action.payload !== state.currentChainId) {
+        state.stage = 'UNINITIALIZED';
+      }
+    });
+  },
+});
+
+const { reducer, actions } = slice;
+export default reducer;
+
+const {
+  disableEnsLookup,
+  processEnsError,
+  processEnsResult,
+  enableEnsLookup,
+  resetResolution,
+} = actions;
+export { resetResolution };
+
+export function initializeEnsSlice() {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const chainId = getCurrentChainId(state);
+    const network = CHAIN_ID_TO_NETWORK_ID_MAP[chainId];
+    const networkIsSupported = Boolean(networkMap[network]);
+    if (networkIsSupported) {
+      ens = new ENS({ provider: global.ethereumProvider, network });
+      await dispatch(enableEnsLookup(network));
+    } else {
+      ens = null;
+      await dispatch(disableEnsLookup());
+    }
+  };
+}
+
+export function lookupEnsName(ensName) {
+  return async (dispatch, getState) => {
+    let state = getState();
+    if (state[name].stage === 'UNINITIALIZED' || ens === null) {
+      await dispatch(initializeEnsSlice());
+    }
+    state = getState();
+    if (
+      state[name].stage === 'NO_NETWORK_SUPPORT' &&
+      !(
+        isBurnAddress(ensName) === false &&
+        isValidHexAddress(ensName, { mixedCaseUseChecksum: true })
+      ) &&
+      !isHexString(ensName)
+    ) {
+      await dispatch(resetResolution());
+    } else {
+      const recipient = ensName.trim();
+
+      log.info(`ENS attempting to resolve name: ${recipient}`);
+      try {
+        const address = await ens.lookup(recipient);
+        await dispatch(processEnsResult(address));
```

Comment:
I might be missing something, but.... `address` is a string right, not an object? But `processEnsResult` appears to expect an object with the property `resolution`.